Ability to queue archive / unarchive for full datasets

bodom0015 opened the pull request:
## Description
<!--- Describe your changes in detail -->

### Problem
The new auto-archiving feature is very nice, but adds the possibility that many files could be accidentally archived en masse. Unarchiving, in this case, would be very tedious as each file needs to be unarchived individually, which requires a lot of UI navigation, clicking, and waiting.

### Approach
Save the user in this case by offering an "Archive All" and "Unarchive All" button at the dataset-level

These buttons will quickly recurse over the contents of the dataset and queue each file inside for archival / unarchival respectively.

I've also added a new Permission for `ArchiveDataset` (since we previously added `ArchiveFile`, we should match that pattern)

### How to Test
Prerequisite: Minio running, Clowder configured to use `S3ByteStorageDriver`, S3 archival extractor configured and running

1. Checkout and run this branch locally
2. Login to Clowder
3. Create a dataset, if you haven't already
4. Within the dataset, upload a file at the root level and create a folder
5. Navigate into the folder you just created, upload a file, and create another folder
6. Navigate once again into this folder and upload one last file
    * You should now have a dataset that looks roughly as follows:
        * Dataset
            * Folder A
                * Folder B
                   * File 3
                * File 2
            * File 1

## Review Time Estimate
<!---When do you want your code reviewed by?-->
- [ ] Immediately
- [x] Within one week
- [ ] When possible

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [x] I have updated the CHANGELOG.md.
- [ ] I have signed the CLA
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

robkooper commented on app/api/Files.scala:
```diff
@@ -1932,7 +1932,7 @@ class Files @Inject()(
       case None => {
         Logger.error("Error getting file " + id)
-        InternalServerError
+        NotFound
```
this is a good change, thanks for fixing this.

robkooper commented:
To fix https://github.com/clowder-framework/extractors-archival/issues/5 can you change the `last_donwloaded` to be now()

bodom0015 commented:
@robkooper this was tricky - since we don't expose a method to directly update the `last_download` date, I've added an optional path to `incrementDownloads`. By default this method will increment both date and count (preserving the current behavior that we see today), but I've added an optional `dateOnly` parameter (false by default) here to allow us to update only the Date of the last download.

This seemed preferable to creating an entire new method that would duplicate half of the logic already present here.
